File: Mixter/Domain/Core/Messages/Events/MessagePublished.cs
using Mixter.Domain.Identity;

namespace Mixter.Domain.Core.Messages.Events
{
    public struct MessagePublished : IDomainEvent
    {
        public MessageId Id { get; private set; }
        
        public UserId Author { get; private set; }

        public string Content { get; private set; }

        public MessagePublished(MessageId id, UserId author, string content)
            : this()
        {
            Content = content;
            Id = id;
            Author = author;
        }

        public object GetAggregateId()
        {
            return Id;
        }
    }
}

File: app/Domain/Messages/MessagePublished.php
<?php

namespace App\Domain\Messages;

use App\Domain\IDomainEvent;

class MessagePublished implements IDomainEvent {

    /**
     * @var MessageId
     */
    private $messageId;

    /**
     * @var string
     */
    private $content;

    /**
     * @param MessageId $messageId
     * @param string $content
     */
    public function __construct(MessageId $messageId, $content) {
        $this->content = $content;
        $this->messageId = $messageId;
    }

    /**
     * @return MessageId
     */
    public function getMessageId()
    {
        return $this->messageId;
    }

    /**
     * @return string
     */
    public function getContent()
    {
        return $this->content;
    }

    public function getAggregateId()
    {
        return $this->getMessageId()->getId();
    }
}